app/code/community/Mbiz/Menu/Block/Page/Html/Topmenu.php

```diff
@@ -38,8 +38,8 @@ public function getCacheLifetime()
     public function getCacheKeyInfo()
     {
         return parent::getCacheKeyInfo() + [
-            'category_id' => $this->getCurrentCategoryId(),
-        ];
+                'category_id' => $this->getCacheByFirstCategory() ? $this->getFirstCategoryId() : $this->getCurrentCategoryId(),
```

Comment:
Can you define the calls `getCacheByFirstCategory` and `setCacheByFirstCategory` into the PHPDoc of the class please?